time/plain_time.mbt

```diff
@@ -0,0 +1,337 @@
+// Copyright 2024 International Digital Economy Academy
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// A time without a time zone
+struct PlainTime {
+  hour : Int
+  minute : Int
+  second : Int
+  nanosecond : Int
+} derive(Eq, Compare)
+
+/// Creates a PlainTime from the hour, minute, second and nanosecond.
+pub fn PlainTime::of(
+  hour : Int,
+  minute : Int,
+  second : Int,
+  nanosecond : Int
+) -> Result[PlainTime, String] {
+  if validate_time(hour, minute, second, nanosecond) {
+    Ok({ hour, minute, second, nanosecond })
+  } else {
+    Err(invalid_time_err)
+  }
+}
+
+/// Creates a PlainTime from a string, like '10:20:30.45678'.
+pub fn PlainTime::from_string(str : String) -> Result[PlainTime, String] {
+  // TODO: better parsing implementation
+  if str.length() < 5 || str.length() > 18 || str[2] != ':' {
+    return Err(invalid_time_err)
+  }
+  let mut hour = 0
+  let mut minute = 0
+  let mut second = 0
+  let mut nanosecond = 0
+  hour = @strconv.parse(str.substring(end=2))?
+  minute = @strconv.parse(str.substring(start=3, end=5))?
+  if str.length() > 7 {
+    if str[5] != ':' {
+      return Err(invalid_time_err)
```

Comment:
If you are using Result, should this line be able to be run and added to coverage?

Reply:
This should be covered by the test case "123:00:00"